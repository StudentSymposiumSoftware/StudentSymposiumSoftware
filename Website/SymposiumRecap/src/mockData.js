/*

  Mock Data has many objects of these fields:
   - Author
   - Title
   - Category
   - Major
   - Professor
   - University
   - Abstract Number

*/


const mock_data = [{ author: 'Elijah Smith',
    professor: 'Olivia Lewis',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Social Sciences and Humanities',
    abstractNumber: 38893888 },
  { author: 'Sophia Campbell',
    professor: 'Mia Lee',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Social Sciences and Humanities',
    abstractNumber: 75957852 },
  { author: 'Lucy Martinez',
    professor: 'Carter Martinez',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'The Sociology of Online Communities in the Digital Age',
    category: 'Natural Sciences',
    abstractNumber: 48203955 },
  { author: 'Zoe Jackson',
    professor: 'Layla Perez',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 64568884 },
  { author: 'Noah King',
    professor: 'William Smith',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Biomedical Sciences',
    abstractNumber: 40679302 },
  { author: 'Joseph Baker',
    professor: 'Aria Thompson',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 37625940 },
  { author: 'Ava Rodriguez',
    professor: 'Victoria Thomas',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Climate Change and Its Effects on Global Food Security',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 35716422 },
  { author: 'Henry Adams',
    professor: 'Lucy Miller',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Business',
    abstractNumber: 82413071 },
  { author: 'Mason Hernandez',
    professor: 'Aiden Perez',
    school: 'UMaine Orono',
    major: 'Business Administration',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 27882383 },
  { author: 'Emma Taylor',
    professor: 'Ethan Hernandez',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Business',
    abstractNumber: 92969314 },
  { author: 'Aurora Martinez',
    professor: 'William Anderson',
    school: 'UMaine Machias',
    major: 'Computer Science',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Business',
    abstractNumber: 29718536 },
  { author: 'Lucas Robinson',
    professor: 'Scarlett King',
    school: 'UMaine Orono',
    major: 'Psychology',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Natural Sciences',
    abstractNumber: 1457768 },
  { author: 'Chloe Adams',
    professor: 'Liam Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Business',
    abstractNumber: 14148682 },
  { author: 'Layla Young',
    professor: 'Alexander Roberts',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Exploring the Feasibility of Space-Based Solar Power',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 60454722 },
  { author: 'Alexander Hall',
    professor: 'Aria Clark',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Education',
    abstractNumber: 41498009 },
  { author: 'Zoe Young',
    professor: 'Amelia Jackson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 23333306 },
  { author: 'Mason Clark',
    professor: 'Scarlett Hernandez',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 97656877 },
  { author: 'Henry Hall',
    professor: 'Ella Garcia',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 91417686 },
  { author: 'Samuel Scott',
    professor: 'Evelyn Green',
    school: 'UMaine Machias',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Allied Health',
    abstractNumber: 4887031 },
  { author: 'Grace Nelson',
    professor: 'Aria Scott',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 36475226 },
  { author: 'Scarlett Hernandez',
    professor: 'Matthew Wright',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 29014475 },
  { author: 'Maya Hernandez',
    professor: 'William Green',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'The Future of CRISPR in Gene Therapy',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 44667664 },
  { author: 'Elijah Rivera',
    professor: 'Harper Martinez',
    school: 'UMaine Orono',
    major: 'Computer Science',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Social Sciences and Humanities',
    abstractNumber: 21709418 },
  { author: 'Liam Taylor',
    professor: 'Ella Walker',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 45822327 },
  { author: 'Mia Robinson',
    professor: 'Noah Green',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'The Impact of Remote Work on Organizational Productivity',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 75710656 },
  { author: 'Evelyn Rodriguez',
    professor: 'Lucas Moore',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Deep Learning for Medical Image Analysis',
    category: 'Allied Health',
    abstractNumber: 39877467 },
  { author: 'Zoe Brown',
    professor: 'Scarlett Thompson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Biomedical Sciences',
    abstractNumber: 74596190 },
  { author: 'Sophie Thomas',
    professor: 'David Rodriguez',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 30556122 },
  { author: 'Maya Mitchell',
    professor: 'Aiden White',
    school: 'UMaine Orono',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Arts',
    abstractNumber: 75180428 },
  { author: 'Jack Anderson',
    professor: 'Harper Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Natural Sciences',
    abstractNumber: 82398661 }]

export const getMockData = () => {
    return mock_data;
}