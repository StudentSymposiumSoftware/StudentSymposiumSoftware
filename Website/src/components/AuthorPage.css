.author-container{
    display: flex;
    flex-direction: column;
    justify-content: start;
    align-items: start;
    width: 95%;
    height: 100%;
}

#author-title{
    font-size: 3vw;
    font-weight: bold;
    padding-bottom: 5vh;
}

#abstract-display{
    display: flex;
    width: 100%;
    justify-content: start;
    align-items: start;
    gap: 40%;
}

.author-subtitle{
    font-size: 1.5vw;
    padding-bottom: 2vh;
}

.hidden{
    display: none;
}

.abstract-list{
    display: flex;
    flex-direction: column;
    justify-content: start;
    align-items: start;
    gap: 1vh;
}

.abstract-item{
    width: 20vw;
    background-color: rgba(255, 255, 255, 0.279);
    color: black;
    
    padding-top: 2.5vh;
    padding-bottom: 2.5vh;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    border-radius: 10px;
    border: solid black 1px;
    transition: all 0.5s ease-in-out;
    user-select: none;
    transform-origin: left;
}

.abstract-item:hover{
    transform: scale(1.05);
}

.abstract-item h3{
    font-size: 0.8vw;
}
#abstract-link{
    cursor: pointer;
}