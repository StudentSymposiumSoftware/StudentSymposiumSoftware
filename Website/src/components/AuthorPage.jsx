import "./AuthorPage.css"
import { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

function AuthorPage(props) {
    const [authorData, setAuthorData] = useState([]);
    const [mentorData, setMentorData] = useState([]);
    const authorName = useParams().name;

    useEffect(() => {
        setAuthorData(props.data.filter(abstract => abstract.author == authorName))
        setMentorData(props.data.filter(abstract => abstract.professor == authorName))
    }, [props.data, authorName])
    
    return (
        <div className="author-container">
            <h2 id="author-title">{authorName}</h2>

            <div id="abstract-display">
                <div className={`${authorData.length > 0 ? "abstract-list" : "hidden"}`} id="authored-abstracts">
                    <h3 className="author-subtitle">Authored:</h3>
                    {authorData.map((abstract, index) => (
                        <div className="abstract-item" key={index}>
                            <h3>{abstract.abstractNumber}</h3> 
                            <h3>{abstract.title}</h3>
                        </div>
                    ))}
                </div>

                <div className={`${mentorData.length > 0 ? "abstract-list" : "hidden"}`} id="approved-abstracts">
                    <h3 className="author-subtitle">Abstracts Approved:</h3>
                    {mentorData.map((abstract, index) => (
                        <div className="abstract-item" key={index}>
                            <h3>{abstract.abstractNumber}</h3>
                            <h3>{abstract.title}</h3> 
                        </div>
                    ))}
                </div>
            </div>
        </div>
    );
}

export default AuthorPage
