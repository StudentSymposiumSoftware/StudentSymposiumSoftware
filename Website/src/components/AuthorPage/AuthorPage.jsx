import "./AuthorPage.css"
import { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

function AuthorPage(props) {
    const [authorData, setAuthorData] = useState([]);
    const [mentorData, setMentorData] = useState([]);
    const authorName = useParams().name;

    useEffect(() => {
        setAuthorData(props.data.filter(abstract => abstract.author == authorName))
        setMentorData(props.data.filter(abstract => abstract.professor == authorName))
    }, [props.data, authorName])
    
    return (
        <div className="author-container">
            <h2 id="author-title">{authorName}</h2>

            <div id="abstract-display">
                {authorData.length > 0 && <div className="abstract-list" id="authored-abstracts">
                    <h3 className="author-subtitle">Authored:</h3>
                    {authorData.map((abstract, index) => (
                        <div className="abstract-item" key={index}>
                            <h3>{abstract.abstractNumber}</h3> 
                            <h3>{abstract.title}</h3>
                        </div>
                    ))}
                </div>}

                {mentorData.length > 0 && <div className="abstract-list" id="approved-abstracts">
                    <h3 className="author-subtitle">Abstracts Approved:</h3>
                    {mentorData.map((abstract, index) => (
                        <div className="abstract-item" key={index}>
                            <h3>{abstract.abstractNumber}</h3>
                            <h3>{abstract.title}</h3> 
                        </div>
                    ))}
                </div>}
            </div>
        </div>
    );
}

export default AuthorPage
