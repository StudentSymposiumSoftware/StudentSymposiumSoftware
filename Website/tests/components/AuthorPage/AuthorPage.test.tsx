import React from 'react';
import { render } from "@testing-library/react";
import { expect, it, vi } from "vitest";
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';

import mockData from '../../../src/mockData';
import AuthorPage from '../../../src/components/AuthorPage';

it('Author Page renders correctly', () => {
    const tree = render(<Router> <AuthorPage data={mockData}/></Router>)
    expect(tree).toMatchSnapshot();
});


it('renders AuthorPage correctly when visiting /author/:name', () => {
    vi.mock('react-router-dom', async (importOriginal) => {
        const actual = (await importOriginal()) as { [key: string]: any };
      
        return {
          ...actual, 
          useParams: () => ({ name: 'Elijah Smith' }), 
        };
    });

    render(
      <Router>
        <Routes>
          <Route path="/author/:name" element={<AuthorPage data={mockData} />} />
        </Routes>
      </Router>
    );
});