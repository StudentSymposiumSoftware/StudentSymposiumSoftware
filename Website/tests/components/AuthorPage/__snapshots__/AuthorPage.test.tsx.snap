// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Author Page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="author-container"
      >
        <h2
          id="author-title"
        >
          Elijah Smith
        </h2>
        <div
          id="abstract-display"
        >
          <div
            class="abstract-list"
            id="authored-abstracts"
          >
            <h3
              class="author-subtitle"
            >
              Authored:
            </h3>
            <div
              class="abstract-item"
            >
              <h3>
                38893888
              </h3>
              <h3>
                Machine Learning Approaches to Financial Fraud Detection
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="author-container"
    >
      <h2
        id="author-title"
      >
        Elijah Smith
      </h2>
      <div
        id="abstract-display"
      >
        <div
          class="abstract-list"
          id="authored-abstracts"
        >
          <h3
            class="author-subtitle"
          >
            Authored:
          </h3>
          <div
            class="abstract-item"
          >
            <h3>
              38893888
            </h3>
            <h3>
              Machine Learning Approaches to Financial Fraud Detection
            </h3>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
